TLM/TLM/Manager/Impl/VehicleRestrictionsManager.cs

```diff
@@ -97,7 +97,7 @@ public class VehicleRestrictionsManager
             ushort nodeId,
             VehicleRestrictionsMode busLaneMode)
         {
-            IDictionary<byte, ExtVehicleType> ret = new TinyDictionary<byte, ExtVehicleType>();
+            IDictionary<byte, ExtVehicleType> ret = new Dictionary<byte, ExtVehicleType>();
```

Comment:
would it not be better to initialize the dictionary with initial capacity?
```
new Dictionary<byte, ExtVehicleType>(50);
```

Reply:
I don't know how fast a dictionary is. Should we make this a hashtable?

Reply:
same thing for the other dictionaries.

Reply:
`Hashtable` is slightly slower than dictionary and because of key and value are stored as `object`, inserting and accessing the data result in boxing/unboxing
I've included test for `Hashset` - iteration is... ~4x worse than Dictionary, not to mention about garbage

![image](https://user-images.githubusercontent.com/19638970/133896398-0ce8a58f-5722-4aff-b768-7948926c9aca.png)

Reply:
what about initial capacity?

> 
> 
> would it not be better to initialize the dictionary with initial capacity?
> 
> ```
> new Dictionary<byte, ExtVehicleType>(50);
> ```

Reply:
In most cases cannot be predicted and dictionary won't have 1000000000 keyvalue pairs, often 1- 8 max.

Reply:
by default it creates 10 entries.
since it is an enum it cannot be greater than 32 (Assuming a 1:1 relationship).
lets leave some space for feature and make it 50 (we can use reflection to count enum member but its slow).

Reply:
Wait, wait... what enum? In this case key-value pair is `laneIndex`(key) and `ExtVehicle`{value}.

Reply:
Ok I see.

Reply:
Funny thing, I was pretty sure that initial capacity of Dictionary is `0`, but then I realized that Rider IDE fetched mscorlib.dll from `.NetFramework 2.0 v2.0.50727` instead of Mono `2.0.50727.1433`, despite the same version, in mono it's `10`, but in .Net is `0` 🤣

Reply:
This is still not a zero garbage code but its much better so its a pass.